arwen/elrondapi/elrondei.go

```diff
@@ -572,6 +597,74 @@ func storageLoad(context unsafe.Pointer, keyOffset int32, dataOffset int32) int3
 	return int32(len(data))
 }
 
+//export setStorageLock
+func setStorageLock(context unsafe.Pointer, keyOffset int32, lockTimestamp int64) int32 {
+	runtime := arwen.GetRuntimeContext(context)
+	storage := arwen.GetStorageContext(context)
+	metering := arwen.GetMeteringContext(context)
+
+	key, err := runtime.MemLoad(keyOffset, arwen.HashLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return -1
+	}
+
+	timeLockKey, err := arwen.CustomStorageKey(context, arwen.TimeLockKeyPrefix, key)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return -1
+	}
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.Int64StorageStore
+	metering.UseGas(gasToUse)
+
+	bigTimestamp := big.NewInt(0).SetInt64(lockTimestamp)
+	return storage.SetStorage(timeLockKey, bigTimestamp.Bytes())
+}
+
+//export getStorageLock
+func getStorageLock(context unsafe.Pointer, keyOffset int32) int64 {
+	runtime := arwen.GetRuntimeContext(context)
+	metering := arwen.GetMeteringContext(context)
+	storage := arwen.GetStorageContext(context)
+
+	key, err := runtime.MemLoad(keyOffset, arwen.HashLen)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return -1
+	}
+
+	timeLockKey, err := arwen.CustomStorageKey(context, arwen.TimeLockKeyPrefix, key)
+	if arwen.WithFault(err, context, runtime.ElrondAPIErrorShouldFailExecution()) {
+		return -1
+	}
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.StorageLoad
+	metering.UseGas(gasToUse)
+
+	data := storage.GetStorage(timeLockKey)
+	timeLock := big.NewInt(0).SetBytes(data).Int64()
+
+	return timeLock
+}
+
+//export isStorageLocked
+func isStorageLocked(context unsafe.Pointer, keyOffset int32) int32 {
+	timeLock := getStorageLock(context, keyOffset)
+	if timeLock < 0 {
```

Comment:
You can delete this condition if `lockTimestamp` is an `uint64`, or if there are useful cases where the timestamp may be negative.

Reply:
Like mentioned above, there is the `-1` case for errors (I followed the same pattern used everywhere, if you think it's better to refactor this, we can)

Reply:
Indeed, `-1` is also returned by `storageStore` and `storageLoad`.